third_party/move/tools/move-decompiler/tests/enum.move

```diff
@@ -0,0 +1,137 @@
+module 0x99::enum_simple {
+    // Note: `Shape` has no `drop` ability, so must be destroyed with explicit unpacking.
+    enum Shape {
+        Circle{radius: u64},
+        Rectangle{width: u64, height: u64}
+    }
+
+    fun destroy_empty(self: Shape) : bool {
+        match (self) {
+            Shape::Circle{radius} => true,
+            Shape::Rectangle{width, height: _} => false,
+        }
+    }
+
+    fun example_destroy_shapes() {
+        let c = Shape::Circle{radius: 0};
+        let r = Shape::Rectangle{width: 0, height: 0};
+        c.destroy_empty();
+        r.destroy_empty();
+    }
+}
+
+
+
+module 0x99::enum_complex {
+    enum Admin has drop {
+        Superuser,
+        User {
+            _0: u64,
+        }
+    }
+    enum Entity has drop {
+        Person {
+            id: u64,
+        }
+        Institution {
+            id: u64,
+            admin: Admin,
+        }
+    }
+    fun admin_id(self: &Entity): u64 {
```

Comment:
I left this comment in a different PR, but it looks like we might want to put the original code here instead of what looks like decompiled code?